gen/enum_test.go

```diff
@@ -399,3 +401,128 @@ func TestEnumAccessors(t *testing.T) {
 		})
 	})
 }
+
+func TestEnumLabelValid(t *testing.T) {
+	tests := []struct {
+		enumItem  te.EnumWithLabel
+		jsonValue string
+	}{
+		{te.EnumWithLabelUsername, `"surname"`},
+		{te.EnumWithLabelPassword, `"hashed_password"`},
+		{te.EnumWithLabelSalt, `"SALT"`},
+		{te.EnumWithLabelSugar, `"SUGAR"`},
+		{te.EnumWithLabelRelay, `"RELAY"`},
+		{te.EnumWithLabelNaive4N1, `"function"`},
+		{te.EnumWithLabel(42), `42`},
+		{te.EnumWithLabel(-1), `-1`},
+		{te.EnumWithLabel(1 << 10), `1024`},
+	}
+	for _, tt := range tests {
+		t.Run(tt.enumItem.String(), func(t *testing.T) {
+			t.Run("JSON round trip", func(t *testing.T) {
+				// marshal
+				label := te.EnumWithLabel(tt.enumItem)
+				b, err := json.Marshal(label)
+				require.NoError(t, err)
+				assert.Equal(t, []byte(tt.jsonValue), b)
+				// unmarshal
+				var gotLabel te.EnumWithLabel
+				err = json.Unmarshal(b, &gotLabel)
+				assert.Equal(t, label, gotLabel)
+			})
+			assertRoundTrip(
+				t, &tt.enumItem,
+				wire.NewValueI32(int32(tt.enumItem)),
+				"test roundtrip "+tt.jsonValue,
+			)
+		})
+	}
+}
+
+func TestEnumLabelInvalidUnmarshal(t *testing.T) {
+	tests := []struct {
+		errVal []byte
+		errMsg string
+	}{
+		{
+			[]byte(`"some-random-str"`),
+			`unknown enum value "some-random-str" for "EnumWithLabel"`,
+		},
+		{
+			[]byte(`"; drop table users;"`),
+			`unknown enum value "; drop table users;" for "EnumWithLabel"`,
+		},
+		{
+			[]byte(`"USERNAME"`),
+			`unknown enum value "USERNAME" for "EnumWithLabel"`,
+		},
+		{
+			[]byte(`"PASSWORD"`),
+			`unknown enum value "PASSWORD" for "EnumWithLabel"`,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(string(tt.errVal), func(t *testing.T) {
+			var expectedLabel te.EnumWithLabel
+			err := json.Unmarshal(tt.errVal, &expectedLabel)
+			assert.Equal(t, err.Error(), tt.errMsg)
```

Comment:
use `assert.EqualError`, otherwise, if it doesn't return an error, this will panic

Reply:
neat! Thanks for the advice